openBCIBoard.js

```diff
@@ -1930,125 +1931,125 @@ function OpenBCIFactory () {
-            this.sync.timeOffsetMaster = math.floor(sum / this.sync.timeOffsetArray.length);
-          } else {
-            this.sync.timeOffsetMaster = this.sync.curSyncObj.timeOffset;
-          }
+      // Add to array
+      if (this.sync.timeOffsetArray.length >= k.OBCITimeSyncArraySize) {
+        // Shift the oldest one out of the array
+        this.sync.timeOffsetArray.shift();
+        // Push the new value into the array
+        this.sync.timeOffsetArray.push(this.sync.curSyncObj.timeOffset);
+      } else {
+        // Push the new value into the array
+        this.sync.timeOffsetArray.push(this.sync.curSyncObj.timeOffset);
+      }
 
-          this.sync.curSyncObj.timeOffsetMaster = this.sync.timeOffsetMaster;
+      // Calculate the master time offset that we use averaging to compute
+      if (this.sync.timeOffsetArray.length > 1) {
+        var sum = this.sync.timeOffsetArray.reduce(function (a, b) { return a + b; });
+        this.sync.timeOffsetMaster = math.floor(sum / this.sync.timeOffsetArray.length);
+      } else {
+        this.sync.timeOffsetMaster = this.sync.curSyncObj.timeOffset;
+      }
 
-          if (this.options.verbose) {
-            console.log(`Master offset ${this.sync.timeOffsetMaster} ms`);
-          }
+      this.sync.curSyncObj.timeOffsetMaster = this.sync.timeOffsetMaster;
 
-          // Set the valid object to true
-          this.sync.curSyncObj.valid = true;
+      if (this.options.verbose) {
+        console.log(`Master offset ${this.sync.timeOffsetMaster} ms`);
+      }
 
-          // Emit it!
-          this.emit('synced', this.sync.curSyncObj);
-          // Save obj to the global array
-          this.sync.objArray.push(this.sync.curSyncObj);
-          // Set to null
-          this.sync.curSyncObj = null;
-          return resolve(rawPacket);
-        })
-        .catch(err => {
-          // Emit the bad sync object for fun
-          var badObject = openBCISample.newSyncObject();
-          badObject.timeOffsetMaster = this.sync.timeOffsetMaster;
-          this.emit('synced', badObject);
-          // Set back to null
-          this.sync.curSyncObj = null;
-          console.log('Error in _processPacketTimeSyncSet', err);
-          return reject(err);
-        });
-    });
+      // Set the valid object to true
+      this.sync.curSyncObj.valid = true;
+
+      // Emit it!
+      this.emit('synced', this.sync.curSyncObj);
+      // Save obj to the global array
+      this.sync.objArray.push(this.sync.curSyncObj);
+
+      let temp = JSON.parse(JSON.stringify(this.sync.curSyncObj));
+      // Set to null
+      this.sync.curSyncObj = null;
+
+      return temp;
+    } catch (err) {
+      console.log(err);
+      // Emit the bad sync object for fun
+      let badObject = openBCISample.newSyncObject();
+      badObject.timeOffsetMaster = this.sync.timeOffsetMaster;
+      this.emit('synced', badObject);
+      // Set back to null
+      this.sync.curSyncObj = null;
```

Comment:
good call Karl, I changed this to not throw and error when it is emitting a bad sync object.